core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -422,6 +425,8 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
       }
 
     if (newAttempts.isEmpty) {
+      logWarning(s"Failed to load application log ${fileStatus.getPath}. " +
```

Comment:
I think this was clearer where it was before. (Which might be the reason for Sean's other comment.)

In fact, I think this log message is scarier than it is useful. An application without an ID is a valid case (it may be still starting up), so printing warnings here is a little misleading.